### PR TITLE
Update dataloader docstrings

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Part of https://github.com/Lightning-AI/lightning/pull/16839

cc @borda

### CODE REVIEW

File: src/lightning/pytorch/core/hooks.py [hunk 1]
```diff
@@ -412,55 +410,15 @@ def train_dataloader(self) -> TRAIN_DATALOADERS:
...
 
         Note:
-            Lightning adds the correct sampler for distributed and arbitrary hardware.
+            Lightning tries to add the correct sampler for distributed and arbitrary hardware.
             There is no need to set it yourself.
-
-        Example::
-
-            # single dataloader
-            def train_dataloader(self):
```
Comment (carmocca) [inline src/lightning/pytorch/core/hooks.py:421]: The LightningDataModule and data sections in the docs already use several examples with a full example such as this MNIST one

So I removed it from these docstring to make it more information-dense

File: src/lightning/pytorch/core/hooks.py [hunk 2]
```diff
@@ -528,44 +461,19 @@ def val_dataloader(self) -> EVAL_DATALOADERS:
...
-
-            # can also return multiple dataloaders
-            def val_dataloader(self):
-                return [loader_a, loader_b, ..., loader_n]
-
         Note:
             If you don't need a validation dataset and a :meth:`validation_step`, you don't need to
             implement this method.
-
-        Note:
```
Comment (carmocca) [inline src/lightning/pytorch/core/hooks.py:560]: Removed these notes because this is now an error: https://github.com/Lightning-AI/lightning/pull/16837

**Review by awaelchli — APPROVED** [review]
LGTM but I would be careful removing all references to dataloader. It is still common and recommended to use DataLoader. It is great that we can pass iterables, but we won't be able to guarantee correctness for distributed training. So for beginners, and the default experience, we should continue to feature DataLoader.

**Comment by carmocca** [comment]
I clarify what is an iterable and how it relates to the DataLoader in #16839

**Review by justusschock — APPROVED** [review]

**Pull request merged**